.github/workflows/lint.yaml

```diff
@@ -0,0 +1,36 @@
+name: Linting
+
+on:
+  pull_request: {}
+  push:
+    branches:
+        - main
+
+permissions: read-all
+
+jobs:
+  compatibility:
+    runs-on: ubuntu-latest
+    steps:
+    - name: Checkout tree
+      uses: actions/checkout@v4
+      with:
+        fetch-depth: 0
+        fetch-tags: true
+
+    - name: Set-up OCaml
+      uses: ocaml/setup-ocaml@v2
+      with:
+        ocaml-compiler: 5.1
+        opam-pin: false
+        opam-depext: false
+        dune-cache: true
+
+    - name: install atddiff
+      run: opam install atd
+
+    - name: atddiff all supported tags
+      shell: bash
+      run: |
+        eval $(opam env)
+          ./scripts/check-backwards-compatability.sh
```

Comment:
Our OCaml development image includes the `atd` package. It's published weekly and the latest is `returntocorp/ocaml:ubuntu-2023-10-17`. This one contains `atddiff` version 2.13.0. It's big but may be faster than `ocaml/setup-ocaml@v2` (I'm not familiar with it).

Reply:
Yeah it's almost certainly faster, but let's leave that for when the features we want are stable (easy change to swap out in a week or two)